agent/client/workflow_shutdown.go

```diff
@@ -21,12 +22,13 @@ func (c *Client) startShutdownListener(ctx context.Context) error {
 		for {
 			resp := proto.ShutdownRequest{}
 			err := stream.RecvMsg(&resp)
-			if err == io.EOF {
+
+			if errors.Is(err, io.EOF) || isCancelledError(err) {
 				return
 			}
 
 			if err != nil {
-				log.Fatal("could not get message from trigger stream: %w", err)
+				log.Fatal("could not get shutdown listener: %w", err)
 			}
```

Comment:
Does forcing a shutdown here in case of an error make sense? Something like using: `os.Exit(-1)`.

Reply:
I believe `log.Fatal` is already doing that

```
// Fatal is equivalent to Print() followed by a call to os.Exit(1).
func Fatal(v ...any) {
	std.Output(2, fmt.Sprint(v...))
	os.Exit(1)
}
```